Tensile/KernelWriter.py

```diff
@@ -309,6 +309,216 @@ def makeSchedule(self, kernel, tensorParametersA, tensorParametersB, localWriteE
+          StartVgprCIdx = self.startVgprReuse
+          kStr += self.comment1("endSummation: remove C-tile [%u, %u) from pool"%(self.startVgprReuse, self.startVgprReuse+max(0, perMfmaDstVgprs-self.startVgprReuse)))
+          self.vgprPool.remove(self.startVgprReuse, max(0, perMfmaDstVgprs-self.startVgprReuse), "ValuC")
+          #update reUse index
+          self.startVgprReuse = self.startVgprReuse + perMfmaDstVgprs
+   
+        for iui in range(kernel["InnerUnroll"]):
+          packA = packCode.findNamedItem("packA_I%s"%(iui))
+          packB = packCode.findNamedItem("packB_I%s"%(iui))
+          # In case localReadDo not generate pack Module
+          # and findNamedItem will return None type
+          # TODO: let all type have pack Module
+          if not packA:
+            packA = Code.Module()
+          if not packB:
+            packB = Code.Module()
+          while packA.items() or packB.items():
+            if packA.items():
+              packAB.addCode(packA.items().pop(0))
+            if packB.items():
+              packAB.addCode(packB.items().pop(0))
+
+        
+        # remove s_nop for packing
+        macIterItems.pop(0)
+
+        iterCode.addCode(syncCode)
+        skipLocalWriteWaitcnt = 0
+        localReadsWaitcnt = 0
+        #i=0 do read for next Tile
+        #i=1..n read for next Tile + movAccvgpr
+        for i in range(numMfmaPerIter):
+          readLeft = readsPerIter
+          accVgprLeft = accMovPerTile
+          # with PrefetchLocalRead, localreads can interleave with mfma
+          packA = Code.Module()
+          packB = Code.Module()
+          packToSched = False
+          movaccVgprToSched = False
+          storeCToSched = False
+
+          if packAB.items():
+             packA.addCode(packAB.items().pop(0))
+             packInstPerMfma = packA.count()
+          if packAB.items():
+             packB.addCode(packAB.items().pop(0))
+             packInstPerMfma += packB.count()
+
+          if packA.items() or packB.items():
+             packToSched = True
+          if readsPerIter !=0:
+             lrdToSched = True
+          if movaccVgprToSched.items():
+             movaccVgprToSched = True
+             
+          if self.numItersPLR:
+            latencyPerMfmaIter = kernel["MatrixInstM"] // 2
+            if packToSched:
+              availableCyclesforRead = latencyPerMfmaIter - (packInstPerMfma*4) - 8
+            else: 
+              availableCyclesforRead = latencyPerMfmaIter - 8
+            numReadsToSched = availableCyclesforRead // 8
+            if packToSched:
+              numReadsToSched = numReadsToSched - packInstPerMfma*4
+            if (i == 0):
+              numReadsToSched  = 4 if (numReadsToSched>4) else numReadsToSched  # cap it to 4 for credit-max 
+            else:
+              numReadsToSched  = 3 if (numReadsToSched>3) else numReadsToSched  # cap it to 3  to allow other instructions
```

Comment:
should if (i==0) be elif? Can this block be combined with the similar if/else block above it?